XenModel/XenAPI-Extensions/VM.cs

```diff
@@ -2073,6 +2073,20 @@ public bool HciWarnBeforeShutdown
                 return other_config != null && other_config.ContainsKey("hci-warn-before-shutdown");
             }
         }
+
+        public bool CanBootOnHost(Host host)

```

Comment:
We don't generally execute API calls in object's properties. This function should be moved into the action class, and use the action's session.